pkg/render/apiserver.go

```diff
@@ -108,69 +100,68 @@ func APIServer(k8sServiceEndpoint k8sapi.ServiceEndpoint,
 			if err != nil {
 				return nil, err
 			}
-			// We only need to add the tlsKeyPair if we created it, otherwise
+			// We only need to add the TLSKeyPair if we created it, otherwise
 			// it already exists.
-			tlsSecrets = []*corev1.Secret{tlsKeyPair}
-			tlsHashAnnotations[TlsSecretHashAnnotation] = rmeta.AnnotationHash(tlsKeyPair.Data)
+			tlsSecrets = []*corev1.Secret{cfg.TLSKeyPair}
+			tlsHashAnnotations[TlsSecretHashAnnotation] = rmeta.AnnotationHash(cfg.TLSKeyPair.Data)
 		}
-		copy := tlsKeyPair.DeepCopy()
+		copy := cfg.TLSKeyPair.DeepCopy()
 		copy.ObjectMeta = metav1.ObjectMeta{
-			Name:      apiServerTLSSecretName(installation.Variant),
-			Namespace: rmeta.APIServerNamespace(installation.Variant),
+			Name:      apiServerTLSSecretName(cfg.Installation.Variant),
+			Namespace: rmeta.APIServerNamespace(cfg.Installation.Variant),
 		}
 		tlsSecrets = append(tlsSecrets, copy)
 	}
 
-	if managementCluster != nil {
-		if tunnelCASecret == nil {
-			tunnelCASecret = voltronTunnelSecret()
-			tlsSecrets = append(tlsSecrets, tunnelCASecret)
+	if cfg.ManagementCluster != nil {
+		if cfg.TunnelCASecret == nil {
+			cfg.TunnelCASecret = voltronTunnelSecret()
+			tlsSecrets = append(tlsSecrets, cfg.TunnelCASecret)
 		}
-		tlsSecrets = append(tlsSecrets, secret.CopyToNamespace(rmeta.APIServerNamespace(installation.Variant), tunnelCASecret)...)
-		tlsHashAnnotations[voltronTunnelHashAnnotation] = rmeta.AnnotationHash(tunnelCASecret.Data)
+		tlsSecrets = append(tlsSecrets, secret.CopyToNamespace(rmeta.APIServerNamespace(cfg.Installation.Variant), cfg.TunnelCASecret)...)
+		tlsHashAnnotations[voltronTunnelHashAnnotation] = rmeta.AnnotationHash(cfg.TunnelCASecret.Data)
 	}
 
 	return &apiServerComponent{
-		installation:                installation,
-		forceHostNetwork:            hostNetwork,
-		managementCluster:           managementCluster,
-		managementClusterConnection: managementClusterConnection,
-		amazonCloudIntegration:      aci,
-		tlsSecrets:                  tlsSecrets,
-		tlsAnnotations:              tlsHashAnnotations,
-		pullSecrets:                 pullSecrets,
-		openshift:                   openshift,
-		k8sServiceEp:                k8sServiceEndpoint,
-		clusterDomain:               clusterDomain,
+		cfg:            cfg,
+		tlsSecrets:     tlsSecrets,
+		tlsAnnotations: tlsHashAnnotations,
 	}, nil
 }
 
+// APIServerConfiguration contains all the config information apiServerComponent needs to render component.
+type APIServerConfiguration struct {
+	K8SServiceEndpoint          k8sapi.ServiceEndpoint
+	Installation                *operatorv1.InstallationSpec
+	HostNetwork                 bool
```

Comment:
Please switch this to ForceHostNetwork, because HostNetwork may still be used even if this field is not set.

Reply:
Done